File: client/src/app/modules/miscellaneous/services/guichet-unique/type.document.service.ts
import { HttpClient, HttpParams } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { TypeDocument } from 'src/app/modules/quotation/model/guichet-unique/referentials/TypeDocument';
import { AppRestService } from 'src/app/services/appRest.service';

@Injectable({
  providedIn: 'root'
})
export class TypeDocumentService extends AppRestService<TypeDocument> {

  constructor(http: HttpClient) {
    super(http, 'quotation/guichet-unique');
  }

  getTypeDocument() {
    return this.getListCached(new HttpParams(), 'type-document');
  }

  addOrUpdateTypeDocument(typeDocument: TypeDocument) {
    this.clearListCache(new HttpParams(), "type-document");
    return this.addOrUpdate(new HttpParams(), "type-document", typeDocument, "Enregistré", "Erreur lors de l'enregistrement");
  }
}

File: src/main/java/com/jss/osiris/modules/quotation/service/guichetUnique/referentials/TypeDocumentService.java
package com.jss.osiris.modules.quotation.service.guichetUnique.referentials;

import java.util.List;

import com.jss.osiris.modules.quotation.model.guichetUnique.referentials.TypeDocument;

public interface TypeDocumentService {
    public List<TypeDocument> getTypeDocument();

    public TypeDocument getTypeDocumentByCode(String code);

    public TypeDocument addOrUpdateTypeDocument(TypeDocument typeDocument);
}
